theano/tensor/opt.py

```diff
@@ -221,89 +221,124 @@ def inplace_elemwise_optimizer(fgraph):
-                            fgraph.replace(r, new_r,
-                                        reason="inplace_elemwise_optimizer")
-                        nb_change_no_validate += 1
-                        if nb_change_no_validate >= check_each_change:
-                            fgraph.validate()
-                            chk = fgraph.checkpoint()
-                            nb_change_no_validate = 0
-                    except (ValueError, TypeError, InconsistencyError) as e:
-                        if check_each_change != 1 and not raised_warning:
-                            print((
-                                    "Some inplace optimization was not "
-                                    "performed due to unexpected error:"), file=sys.stderr)
-                            print(e, file=sys.stderr)
-                            raised_warning = True
-                        fgraph.revert(chk)
-                        continue
-                    candidate_inputs.remove(candidate_input)
-                    node = new_node
-                    baseline = inplace_pattern
-                    break
+                            new_node = new_outputs[0].owner
+
+                            # Add changed node into changed_node_since_validate
+                            if node not in changed_node_since_validate:
+                                changed_node_since_validate.append(node)
+
+                            for r, new_r in zip(node.outputs, new_outputs):
+                                fgraph.replace(r, new_r,
+                                            reason="inplace_elemwise_optimizer")
+                            nb_change_no_validate += 1
+
+                            if nb_change_no_validate >= check_each_change:
+                                fgraph.validate()
+                                chk = fgraph.checkpoint()
+                                changed_node_since_validate = []
+                                nb_change_no_validate = 0
+                        except (ValueError, TypeError, InconsistencyError) as e:
+                            if check_each_change != 1 and not raised_warning:
+                                print((
+                                        "Some inplace optimization was not "
+                                        "performed due to unexpected error:"), file=sys.stderr)
+                                print(e, file=sys.stderr)
+                                raised_warning = True
+                            fgraph.revert(chk)
+
+                            # Add changed node to failed_list
+                            if len(changed_node_since_validate) > 1:
+                                failed_list.extend(changed_node_since_validate)
+                                changed_node_since_validate = []
+                            # Add failed node to failed_list
+                            if node not in failed_list:
```

Comment:
This could be the cause of the big slow down. Here node was already in changed_node_since_validate. So this do much comparison for nothing as the condition will always be False.